File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    q1_sum, q2_sum = sum(queue1), sum(queue2)
    q1, q2 = deque(queue1), deque(queue2)
    q_len = 2 * len(queue1)
    
    if q1_sum == q2_sum:
        return 0
    if (q1_sum + q2_sum) % 2 != 0:
        return -1
    
    answer = 0
    while answer < 2 * q_len:
        if q1_sum < q2_sum:
            num = q2.popleft()
            q1.append(num)
            q1_sum += num
            q2_sum -= num
            answer += 1
        elif q1_sum > q2_sum:
            num = q1.popleft()
            q2.append(num)
            q1_sum -= num
            q2_sum += num
            answer += 1
        else :
            return answer
    
    
    return -1

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.js
function solution(queue1, queue2) {
    let answer = 0;
    let queue = [...queue1, ...queue2]
    let qSum1 = queue1.reduce((pre, num) => pre + num),
        qSum2 = queue2.reduce((pre, num) => pre + num);
    let idx1 = 0, idx2 = queue2.length;
    let len = queue1.length * 2
    let temp;

    if ((qSum1 + qSum2) % 2 != 0)
        return -1

    while(answer <= len * 2) {
        if (qSum1 === qSum2)
            return answer;

        if(qSum1 > qSum2){
            temp = queue[idx1];
            idx1 = (idx1 + 1) % len;
            qSum1 -= temp; 
            qSum2 += temp;
            answer++;
        }
        if(qSum1 < qSum2){
            temp = queue[idx2];
            idx2 = (idx2 + 1) % len;
            qSum1 += temp; 
            qSum2 -= temp;
            answer ++;
        }
    }
    return -1;
    
}